internal/hcsoci/create.go

```diff
@@ -105,18 +105,13 @@ func CreateContainer(createOptions *CreateOptions) (_ *hcs.System, _ *Resources,
 		}
 	}
 
-	// Create a network namespace if necessary.
+	// Attach networking
 	if coi.Spec.Windows != nil &&
 		coi.Spec.Windows.Network != nil &&
 		schemaversion.IsV21(coi.actualSchemaVersion) {
 
 		if coi.NetworkNamespace != "" {
 			resources.netNS = coi.NetworkNamespace
-		} else {
-			err := createNetworkNamespace(coi, resources)
-			if err != nil {
-				return nil, resources, err
-			}
 		}
 		coi.actualNetworkNamespace = resources.netNS
```

Comment:
I think you need this part in the `if coi.NetworkNamespace != ""` now because `createNetworkNamespace` used to set it.

Reply:
This code was already there in hcsshim, before I added the support for containerd namespace. I assumed this was kept to support container creation using v2 schema in RS4